server/api/social/link.js

```diff
@@ -0,0 +1,114 @@
+import { Router } from 'express';
+import { jwtAuth } from '../../util/jwt';
+import { ValidationError } from '../../../util/ValidationHelper';
+
+const { userCollection: dbRef } = require('../../util/firebase');
+
+const router = Router();
+
+router.post('/social/links/new', jwtAuth, async (req, res, next) => {
+  try {
+    const {
+      user,
+      link,
+    } = req.body;
+    if (req.user.user !== user) {
+      res.status(401).send('LOGIN_NEEDED');
+      return;
+    }
+
+    const {
+      id,
```

Comment:
what happens when i hijack payload `id = facebook` ? will I be enable to make a trap facebook button in my profile?
Handling create id logic pure serverside seems better?